c2corg_ui/views/document.py

```diff
@@ -413,20 +421,43 @@ def _get_api_cache_key_from_etag(self, etag):
     def _get_response(self, page_html):
         return get_response(self.request, page_html)
 
+    def _redirect(self, id, lang, slug=None):
+        location = ''
+        if slug is None:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view_id_lang', id=id, lang=lang)
+        else:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view', id=id, lang=lang, slug=slug)
+        raise HTTPMovedPermanently(location=location)
```

Comment:
> When requesting /waypoints/1234 (without lang), the redirect depends on the current interface language.

OK, makes sense. So we need a 302 when the lang is missing and 301 if it is provided :)

> Second problem: If redirects_to is set for a document, a permanent redirect to the linked document will be made. But redirects_to can be changed!

Why would it change? This case has never happened in v5 since 2007.

My concern is that we need to make sure that redirecting documents are not referenced by search engines and that the target document is always taken into accound instead. It seems we need the 301 for this, doesn't it?